polkadot/xcm/xcm-executor/src/lib.rs

```diff
@@ -923,7 +928,10 @@ impl<Config: config::Config> XcmExecutor<Config> {
 				.as_mut()
 				.ok_or(XcmError::BadOrigin)?
 				.append_with(who)
-				.map_err(|_| XcmError::LocationFull),
+				.map_err(|e| {
+					log::error!(target: "xcm::process_instruction::descend_origin", "Failed to append to Location: {:?}", e);
```

Comment:
little bit misleading, because `append_with` returns `who` on error 
```suggestion
					log::error!(target: "xcm::process_instruction::descend_origin", "Failed to append junctions: {e:?}");
```